### PR TITLE
Extract video src from children

### PR DESCRIPTION
**Pull request by GabiThume**
Extract src from children of video source tags, by creating an Object like
```
video =
  "video/mp4": "foo.mp4"
  "video/webm": "foo.webm"
```
and using only `video: "foo.mp4"` for now.

### CODE REVIEW

**Comment by GabiThume** [comment]
@bergie it looks like one possibility for extracting
```
<video>
  <source type="video/mp4" src="foo.mp4"/>
  <source type="video/webm" src="foo.webm"/>
</video>
```
is to use the video key as an Array type, returning
```
video = [
  'foo.mp4',
  'foo.webm'
]
```
or to use it with types as keys
```
video =
  "video/mp4": 'foo.mp4'
  "video/webm": 'foo.webm'
```
what do you think?

**Comment by bergie** [comment]
@GabiThume currently the schema only allows one video, so first step would be to pick the "best one" (mp4?). We could tweak schema down the line

File: index.coffee [hunk 1]
```diff
@@ -211,7 +211,22 @@ module.exports = class Flatten
         video =
           type: 'video'
           html: @tagToHtml tag, id
-        video.video = tag.attribs.src if tag.attribs.src
+        if not tag.attribs.src
+          sources = {}
+          for child in tag.children
+            if child.name is "source" and child.attribs?.src?
+              normalized = @normalizeUrl child.attribs.src, id
+              console.log child.attribs
```
Comment (bergie) [inline index.coffee]: Please remove the log

File: index.coffee [hunk 2]
```diff
@@ -211,7 +211,22 @@ module.exports = class Flatten
...
+              if child.attribs.type?
+                sources["#{child.attribs.type}"] = normalized
+          if sources["video/ogv"]
+            video.video = sources["video/ogv"]
+          else if sources["video/mp4"]
+            video.video = sources["video/mp4"]
+          else
+            video.video = sources[Object.keys(sources)[0]] if (Object.keys(a)).length
+        else
```
Comment (bergie) [inline index.coffee]: Instead of an `else`, this might be better as `if video.attribs.src`

**Review by bergie — CHANGES_REQUESTED** [review]
At least remove the logger

**Comment by GabiThume** [comment]
@bergie can you please review this again? :)

**Review by bergie — APPROVED** [review]

**Pull request merged**